### PR TITLE
chore: switch applitools to percy for website VRT

### PR DESCRIPTION
**Pull request by SiTaggart**
# In this PR

## Replace Applitools with Percy

- Replaces the VRT tool we use to Percy, using Cypress as the test runner still
- Configure Percy to hide iframes to prevent flakey Snapshots
- Configure Percy to remove scroll behaviour and overflow from site body so we can screenshot an entire page
- Create two new scripts to run Cypress via Percy CLI to run VRT locally
- Remove the need to remove the VRT label from PRs as it's no longer necessary
- Update all docs regarding Applitools for Percy.
- Introduce new environment variable `PERCY_TOKEN`
- Rename `USE_CYPRESS_EYES` to be more generic `USE_CYPRESS_VRT`
- Genericise the Eyes cypress code to be applicable to any VRT tool in the future.

### CODE REVIEW

**Review by shleewhite — APPROVED** [review]
Looks good to me! Can we do a brownbag or something with the team to go over the new tool? Would be helpful to go over how to approve changes and go through a PR together. Applitools had that weird thing where you approved the snapshots and then had to go to another link to merge the changes into main.

File: .percy.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const percyCSS = `
+/* disable all the iframes in the docs site */
+iframe {
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply .percy.js:3]: Yes, yes it will 👍🏼

**Comment by SiTaggart** [comment]
> Can we do a brownbag or something with the team to go over the new tool? Would be helpful to go over how to approve changes and go through a PR together

Yeah, I can do a loom and a quick brownbag after it's merged it. I still have some open questions myself on how to create the baseline on `main`, but that's hard to know the answers to without having a baseline on main created after this PR.

File: .github/workflows/remove_vrt_label.yml [hunk 2]
```diff
@@ -1,15 +0,0 @@
-name: Remove VRT Label
```
[the remark this answers is not in the document]
Comment (shleewhite) [reply .github/workflows/remove_vrt_label.yml:1]: Yah, that makes sense!

**Review by shleewhite — APPROVED** [review]

File: .github/workflows/on_pull_request_cypress.yml [hunk 3]
```diff
@@ -82,5 +96,9 @@ jobs:
           parallel: true
           group: "Paste Actions with VRT Parallel"
           config: baseUrl=${{ steps.waitForDeployment.outputs.url }}
+          # enabled percy to run cypress for VRT
+          command-prefix: "percy exec --parallel -- npx"
         env:
-          USE_CYPRESS_EYES: true
+          USE_CYPRESS_VRT: true
+          PERCY_PARALLEL_NONCE: ${{ needs.percy-batch-id.outputs.result }}
+          PERCY_PARALLEL_TOTAL: 3
```
Comment (TheSisb) [inline .github/workflows/on_pull_request_cypress.yml:104]: question: why 3? Is bigger better here?

File: internal-docs/engineering/doc-site/visual-regression-testing.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+3. Run selected tests by clicking on their names in the GUI
+
+> 👉 **Tip**: In order for changes to the environment variables to take effect, you must re-start cypress. 👈
+
+## CI workflow
+
+Below is a flow chart illustrating possible possible results of the VRT run and actions that should be taken.
+
+```mermaid
+graph TD
```
Comment (TheSisb) [inline internal-docs/engineering/doc-site/visual-regression-testing.md:66]: praise: very nice

**Review by TheSisb — APPROVED** [review]

File: .github/workflows/on_pull_request_cypress.yml (hunk 3 above)
Comment (SiTaggart) [reply .github/workflows/on_pull_request_cypress.yml:104]: Ok, so it needs to match the number of machines you parallelise on. So we split the cypress job across three machines, we need to tell Percy to expect 3 "I'm done" events so it knows when to start processing

Comment (TheSisb) [reply .github/workflows/on_pull_request_cypress.yml:104]: Good to know

**Pull request merged**